torch/_dynamo/polyfill.py

```diff
@@ -35,3 +37,11 @@ def repeat(item, count):
 
 def radians(x):
     return math.pi / 180.0 * x
+
+
+def accumulate_grad(x, new_grad):
+    new_grad = torch.clone(new_grad)
+    if x.grad is None:
+        x.grad = new_grad
+    else:
+        x.grad.add_(new_grad)
```

Comment:
This is a simplification, are you planning to do the more complicated version?

Reply:
I am but not in this PR. This is to unblock compiled ddp work. 

I need some more understanding and write new test cases with compiled_autograd to do a full fidelity solution. Maybe at the end of this week or next week.

Reply:
I guess the main trouble is that you are potentially changing the behavior; possibly evenly silently incorrect if x.grad is broadcasted and you need to un-broadcast it before adding the new grad in.

Reply:
I believe the grad_sum_to_size happens in compiled autograd not accumulate_grad_.  Though there is a lot of logic regarding layout this is missing.